joern-cli/frontends/c2cpg/src/main/scala/io/joern/c2cpg/parser/ParseProblemsLogger.scala

```diff
@@ -27,15 +27,17 @@ trait ParseProblemsLogger {
   /** The exception message might be null for parse failures due to ambiguous nodes that can't be resolved successfully.
     * We extract better log messages for this case here.
     */
-  protected def extractParseException(exception: Throwable): String = Option(exception.getMessage) match {
-    case Some(message) => message
-    case None =>
-      exception.getStackTrace
-        .collectFirst {
-          case stackTraceElement: StackTraceElement if stackTraceElement.getClassName.endsWith("ASTAmbiguousNode") =>
-            "Could not resolve ambiguous node!"
-        }
-        .getOrElse(exception.getStackTrace.mkString(System.lineSeparator()))
+  protected def extractParseException(exception: Throwable): String = {
+    Option(exception.getMessage) match {
```

Comment:
`Option(exception.getMessage).getOrElse { ... }` would be more readable IMO - just a thought